src/plugins/shared-variables/drawing/variable-object.tsx

```diff
@@ -1,51 +1,66 @@
+    type: typeField("variable"),
+    variableId: types.string
+  })
+  .views(self => ({
+    get boundingBox() {
+      const {x, y} = self;
+      // FIXME: this is a problem. The model is looking at the rendered element.
+      // Also MobX might not be able to cache this correctly.  
+      // We only really know the size once the text is rendered. 
+      // A possible solution is to add volatile props for chipWidth and chipHeight and then have
+      // a size observer on the component so it updates these props on the model.
+      // Another solution is to somehow move the boundingBox calculation into the component so
+      // it can work with its ref. 
+      const chipWidth = document.getElementById(self.variableId)?.getBoundingClientRect().width || 75;
+      const chipHeight = document.getElementById(self.variableId)?.getBoundingClientRect().height  || 24;
+      const nw: Point = {x, y};
+      const se: Point = {x: x + chipWidth, y: y + chipHeight};
+      return {nw, se};  
+    }
+  }));
+export interface VariableChipObjectType extends Instance<typeof VariableChipObject> {}
+export interface VariableChipObjectSnapshot extends SnapshotIn<typeof VariableChipObject> {}
 
-export class VariableObject extends DrawingObject {
-  declare model: VariableDrawingObjectData;
-  drawingContent: DrawingContentModelType;
-
-  constructor(model: VariableDrawingObjectData, drawingContent: DrawingContentModelType) {
-    super(model);
-    this.drawingContent = drawingContent;
-  }
+export const VariableChipComponent: React.FC<IDrawingComponentProps> = function ({model, handleHover, drawingContent}){
+  if (model.type !== "variable") return null;
+  const { id, x, y, variableId } = model as VariableChipObjectType;
 
-  public getBoundingBox() {
-    const {x, y} = this.model;
-    const chipWidth = document.getElementById(this.model.variableId)?.getBoundingClientRect().width || 75;
-    const chipHeight = document.getElementById(this.model.variableId)?.getBoundingClientRect().height  || 24;
-    const nw: Point = {x, y};
-    const se: Point = {x: x + chipWidth, y: y + chipHeight};
-    return {nw, se};
+  const selectedVariable = findVariable(drawingContent, variableId);
+  if (!selectedVariable) {
+    return null;
   }
 
-  public render(options: DrawingObjectOptions): JSX.Element|null {
-    const {x, y, variableId } = this.model;
-    const {id, handleHover} = options;
-
-    const selectedVariable = findVariable(this.drawingContent, variableId);
-    if (!selectedVariable) {
-      return null;
-    }
+  return (
+    <foreignObject
+      key={id}
+      x={x}
+      y={y}
+      overflow="visible"
+      onMouseEnter={(e) => handleHover ? handleHover(e, model, true) : null }
+      onMouseLeave={(e) => handleHover ? handleHover(e, model, false) : null }
+    >
+      { // FIXME: the same variable can have multiple chips so that would mean multiple
+        // spans with the same id. The model.id would be a better id to use.
+      }
```

Comment:
Seems like if this isn't going to be fixed in this PR there should be a PT story to fix it, given that it presumably results in a confusing bug.

Reply:
I made a story for it: https://www.pivotaltracker.com/story/show/182352893 I'll put that story URL in the comment above too.